src/openxr/extensions/xr_ext_hand_tracking_extension_wrapper.cpp

```diff
@@ -194,7 +194,8 @@ void XRExtHandTrackingExtensionWrapper::update_handtracking() {
 		return;
 	}
 
-	const XrTime time = openxr_api->get_frame_state().predictedDisplayTime;
+	const XrTime time = openxr_api->get_frame_state().predictedDisplayTime + openxr_api->get_frame_state().predictedDisplayPeriod; // predict for next frame, that is when we use this info
```

Comment:
Same comment here about updating the variable's name.

It may be more flexible to add a utility method that returns the next frame time and have all three locations call it.
As you mention above, doing so may not be needed for mono applications, so having a centralized location would allow to update the returned `XrTime` value based on the application type.

Reply:
Love the idea for a function so that's what I end up doing, also checked for our mono condition even though there currently are no use cases for it.